tests/delete/tests.py

```diff
@@ -606,6 +606,13 @@ def receiver(instance, **kwargs):
 
 class FastDeleteTests(TestCase):
 
+    def test_fast_delete_all(self):
+        with self.assertNumQueries(1) as ctx:
+            User.objects.all().delete()
+        sql = ctx.captured_queries[0]['sql']
+        # No subqueries should be used when performing a full delete.
+        self.assertEqual(sql.count('SELECT'), 0)
```

Comment:
I'd've used `assertNotIn('SELECT', sql)`

Reply:
Works for me.